botocore/client.py

```diff
@@ -307,9 +332,27 @@ def __init__(self, serializer, endpoint, response_parser,
         self.meta = ClientMeta(event_emitter, self._client_config,
                                endpoint.host, service_model,
                                self._PY_TO_OP_NAME)
+        self._validate_arguments()
+        self._register_handlers()
+
+    def _validate_arguments(self):
+        if self.meta.s3_addressing_style not in ['virtual', 'default']:
```

Comment:
I was thinking that as well. I was concerned that you can also supply it through the config file it would not hit this validation, but  we currently always create a new config object when creating a new client.